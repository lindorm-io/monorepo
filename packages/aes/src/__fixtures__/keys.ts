import { Kryptos, KryptosFromB64 } from "@lindorm/kryptos";

const defaults = {
  notBefore: new Date("2023-01-01T01:00:00.000Z"),
  updatedAt: new Date("2024-01-01T01:00:00.000Z"),
  expiresAt: new Date("2024-03-01T00:00:00.000Z"),
  issuer: "https://test.lindorm.io/",
  jwksUri: "https://test.lindorm.io/.well-known/jwks.json",
};

const EC: KryptosFromB64 = {
  ...defaults,
  id: "9e568fdb-ed87-5a24-800a-44e1a143d5da",
  curve: "P-521",
  createdAt: new Date("2023-01-01T00:00:00.000Z"),
  privateKey:
    "MIHuAgEAMBAGByqGSM49AgEGBSuBBAAjBIHWMIHTAgEBBEIATAYo3DQYroDV5EgJSUs_kOIvnEScZen73gXa5oQkub0ekQmgOJdPQjINUsdYRn67QK_oBdNUhVbtG_qdxqIgarehgYkDgYYABACN2PIVpRTdfXLmNxkg8Bk2m5netqYsNW2Lefhklr2jfJiVUJUDPoZoGfabGzHEgsKjP2HRbPsEI_tND3x4N9VW0QAID2UYXa7GN0izHWIFRdjVYuR5-0jywFtd-o_N2POdrvlV8xumdVK-TiSPEIdfKoL_Iu0e7IKTsJsj-UmE8rDJnw",
  publicKey:
    "MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQAjdjyFaUU3X1y5jcZIPAZNpuZ3ramLDVti3n4ZJa9o3yYlVCVAz6GaBn2mxsxxILCoz9h0Wz7BCP7TQ98eDfVVtEACA9lGF2uxjdIsx1iBUXY1WLkeftI8sBbXfqPzdjzna75VfMbpnVSvk4kjxCHXyqC_yLtHuyCk7CbI_lJhPKwyZ8",
  type: "EC",
};

export const TEST_EC_KEY = Kryptos.from("b64", {
  ...EC,
  algorithm: "ECDH-ES",
  operations: ["encrypt", "decrypt"],
  use: "enc",
});

const OCT: KryptosFromB64 = {
  ...defaults,
  id: "c124f42f-5a61-582d-9fdd-b834de3336a2",
  createdAt: new Date("2023-01-01T00:10:00.000Z"),
  privateKey: "NjVhNF9EK0BTUWJAd0AwUU9IQWkmdTgwWEo2QDJmYjI",
  type: "oct",
};

export const TEST_OCT_KEY = Kryptos.from("b64", {
  ...OCT,
  algorithm: "dir",
  operations: ["encrypt", "decrypt", "deriveKey"],
  use: "enc",
});

const RSA: KryptosFromB64 = {
  ...defaults,
  id: "8bb9bd86-2e70-5cd0-a1cc-b60a5eb6d574",
  createdAt: new Date("2023-01-01T00:30:00.000Z"),
  privateKey:
    "MIIJKQIBAAKCAgEApq21wR1t9ZnC0R2vgmUrDGjq0G5h5THrtC4kdKfavu2t7iwV0RHsQCJU_uboan63dQysFlYLrqvUluMKBgb8lNAGRc1M7LvGKh3zQlV1gtkh71oVlRqUpuGvY6FbBSCVZThC3oRXSlKpRXSMuHY_ohmsXSkZCDReTZevLmzTG7r_bDp8mK05moNU2dfBhh7KtTssyp6etufDrpxsBXF2i-2VzhwMX2GZJY_7DouXgHWTsLIOLqH18gNMUU8RYULNqiA8vBsubVWV9ygr6wLxdZslm4yAXfMVPj6dQ7FCwo4L7gxtqK3M5Olv1C7rRi0NpTCFuUZlOnufWcrSToPKRDvAFUkZPyk9qzyiww5ckVW3ZRaPpVF36RJN-VEU2dskfw2Md2NrMV3W8LNhXryU004CSjJy6bqmer2cQ9UwbZR8y9n4Vxf45lntZMaZ3ZOYnaXuHZL-j3AcZ82bRsApcQneRhJHElA-nJicuZ4s22-B_rvOBu8YNOR5PutAGZz7J5hGr8L-wGDHktqG0ChY30RRlbvK94XitwzXljhBeDmf5SRRoUVbba8PcF91c7IHlb5q4wWxOgEXM9kb0_Oik6iXCGfZVog8DB8uYI6P2Ta5pvMY_Jt_5U-ULu3P8MPNsfHtY0n1Kpf6MP6srdz2kZupT9VL_mReSv3korv7XHMCAwEAAQKCAgBSyOuBt6LCPfUHlSVzb9kiSEf8bx67TgCiVNqxgB9C_l0JcycA2NSNBCBQHJLqNfbbsYgVbfZOcrTbnnd_h9_S78F10OK4oAjsKUUFuNltPHhA3-7hSDr16hkJNH63PLhG9lssFvIeDZFaWGl-QukXE4ZLHRr8Orpt8n1FYzK1TrsHX8M7t6npJvqxWhqaCVgoDHSSTJdLBDDqXHHAgFrfZAkoPkEc7KY6hFkvH8EHQh6xP7aQjcvQwLeamNlqg9oPegSAb0WvczdRc0fg6WzRO1ZB_FZTk2UFLedBVFzysceWG-EpSu198r2i1wzvJXhmCAQbAi5MDlOWrwimi4mdbejnH_YbYJPvAnld5kpUgAvF3p-plAHaMHDA1hp8GrPcPh9QH5gmy5XEbdN7aV6ekAhMRIczxfWnzOc21l3nemK2bpGlSequBzQsAteYk5jYVBYOnfcTwzhS5nS1LPGJ_AVJEdyvJeM3C1WfC-nNuIdBPEvaQzeD8Re9-IY9Az_x_jW9HzeOzXfIkuNfqSyQeOmgUBIQRvyRg1Xfh0KQGp4Ihf2IcqajOo8-D8ZEUvHaheyEHdRXrORH0BFLIQ8UqmC4XjVOx7q3FURgXuqtFoGDx8cGvfHTLKozWrRhtS9DKCeEk-oGKBrv8cYF0-oebbSm2538Dji2XeobJWyldQKCAQEA13wEi30WTdjeMXVWUfgJkFfcApRMnXQHmtNDzS30qao96YNC5yPzrZyfFFhUoOWCi5fmOAMdMtS7BL6ZJp9aroMAi8IhccYdt8rEHxdbMR5pbQ_TqsH1wYW64ETFViR9rPsg5b8CvjmPxW-1FMwl5uOwmrC7sxUrCSTk1PglZVj1ALR2Z1rUjBKQLQFsh1iFBZepxGGD3kbeVqt04d_lYPSewyLGACvSx1BLbR5y8KBNVSialbb9n2sFKSL6gJytr-dHAMdPIM9SQGawrym0ZiEK5FxYsQUaVP5ulw0FrjzwafXBkFjl2FUjwFE3bnRhaR8BGbJAD8xqZarYSzhEHQKCAQEAxgSAqhB71wfDx57gxT8dnTzKNNsklBrMA3Cl3Zip2EG4L1pS1Qd_mL9LjIfF-zgWTMbGW_SQhoi1Fq4MSUFw5R30HP1qbcHw__WnzqLHYVRpJvIQGfe3LRx-EijblfAmKRWUsbvkn-bpwHrrJSm41I5jjmvam366t6NWLBiar4YAMZ-HTSv5-rJkJMsqGCRZHdregCMGSauIXddDZv-gr9OZnpQRX8CTe72tNdkOjGrknB6gxWJxcV3wU08uwUVwfiXDbqmT8SfFEtpAI_SKxXqDWCjJpKT-cxC5QesHapNEHQWO7M0MfsytejU6QmNTESRlb6P8Y5vlcTWMG8kdzwKCAQEAr2Tizici4HqFYPyBJnceUb8x9WeAdSjch8kuhvPBEdjgTBHrgvhggA_b7FN5jv1BJR6e5Tijq5w8HcCGKEnNJf67kwaQ8mqwBcmlmSkJsRFv_rXDNFpsDvkrAyD2_wb0xLgkLt0q_msb-zjRy_vfXfA277KrFTlFrku5dyAEUFn4EnrRZY8pMlppwDhIfmIu5D5ukMpO674De8hTewcjTafPEkZbpAgHqPRScJ84L2RLjeVe7j3kcr5JlRJDlCvztBosURXp8mrOKF1tsyXfEMIUe4wvJOdiZ4wxuAGBfjFb_uNqinPRoiSq1P9Gf0F7iHUUEDZmrnVUpSxKoe9KgQKCAQA4kJa0feryXPx3LJJmpgXuoavdd3A2goUewnIPIAn8haF4nWOmCPNRDewl468ReU8ddXa6U91pIOoh5WwLDQ-T03-lPFrHqawPYfwiketEiVlUT7d7w4Bm-Yml5Ff9t9Ee0VRdhR_k8ZB6dGJxhj6PMXGNAJA89PGffaAjDpA5bZzmj_yVded9zrG-tO58YP24Aoqmrdqg9sgIjnJpnMIpF9f9Lazc3mUC5wbwp-V5BRvy4OjqGZnwWXZKRgdEHHLrBhySowxoygSIwyGpY7S0fq_AYj5fsuYgtnlFND81GYZOiATw7g4QtqUjkMropfjXn77-Kijxx1m-gh5Y3ZWPAoIBAQCOmEBewD34QEnOioz0fZCzWnzd718k-ac7jbjbmTDM6dD6ZGS0RiG-6VfGqPy2D5wJvd3_pV68urq9pCcA6YErcXsXnRuGu1dX7Qu0-D-eHL7XsXYovzd3lbhfp31KP2wnbhwMnp81FO1wrHY9FvRHmIvWd7OE4z8JWDlspLJVqIFgcqOZJCHBYbz32nY8qkr2KhneJPJ4XibBje5xvRTiF_qTPAa633peDgoVCHkg8Ub1b6X9vg_VXSDeJy3EXlRYEGlfuhEfkSSg9oAYuI-XqnCGqj9JffrncWMq7TY5jduveOAyAZwk0FFsHrius3JUoV-aXlueJYKEFG0MmMcP",
  publicKey:
    "MIICCgKCAgEApq21wR1t9ZnC0R2vgmUrDGjq0G5h5THrtC4kdKfavu2t7iwV0RHsQCJU_uboan63dQysFlYLrqvUluMKBgb8lNAGRc1M7LvGKh3zQlV1gtkh71oVlRqUpuGvY6FbBSCVZThC3oRXSlKpRXSMuHY_ohmsXSkZCDReTZevLmzTG7r_bDp8mK05moNU2dfBhh7KtTssyp6etufDrpxsBXF2i-2VzhwMX2GZJY_7DouXgHWTsLIOLqH18gNMUU8RYULNqiA8vBsubVWV9ygr6wLxdZslm4yAXfMVPj6dQ7FCwo4L7gxtqK3M5Olv1C7rRi0NpTCFuUZlOnufWcrSToPKRDvAFUkZPyk9qzyiww5ckVW3ZRaPpVF36RJN-VEU2dskfw2Md2NrMV3W8LNhXryU004CSjJy6bqmer2cQ9UwbZR8y9n4Vxf45lntZMaZ3ZOYnaXuHZL-j3AcZ82bRsApcQneRhJHElA-nJicuZ4s22-B_rvOBu8YNOR5PutAGZz7J5hGr8L-wGDHktqG0ChY30RRlbvK94XitwzXljhBeDmf5SRRoUVbba8PcF91c7IHlb5q4wWxOgEXM9kb0_Oik6iXCGfZVog8DB8uYI6P2Ta5pvMY_Jt_5U-ULu3P8MPNsfHtY0n1Kpf6MP6srdz2kZupT9VL_mReSv3korv7XHMCAwEAAQ",
  type: "RSA",
};

export const TEST_RSA_KEY = Kryptos.from("b64", {
  ...RSA,
  algorithm: "RSA-OAEP-256",
  operations: ["encrypt", "decrypt"],
  use: "enc",
});
