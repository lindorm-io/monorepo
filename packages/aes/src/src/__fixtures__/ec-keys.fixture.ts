import { Kryptos } from "@lindorm/kryptos";

export const EC_KEY_SET = Kryptos.from("pem", {
  id: "b1ca2923-7a82-5f21-8d23-8d76cf4e2abf",
  curve: "P-521",
  privateKey:
    "-----BEGIN PRIVATE KEY-----\n" +
    "MIHuAgEAMBAGByqGSM49AgEGBSuBBAAjBIHWMIHTAgEBBEIB+1dzKp5FO/wISTaI\n" +
    "PHCv1yZLMZl0o6BAXgMKibC6e8iFI9W4YIrZYTInHDpqA/wNd764XikWWse9jRRw\n" +
    "KLYy02KhgYkDgYYABAFyw1SdEKD9Sz2XojfjqCPVoilq5YSZjLBmZtXG2x6ydRUW\n" +
    "3Q/IL7kfEtKycJ42z9UhP4adevymNBQ4J02/Yx/jswGcKMep28r825kUE64LbsUc\n" +
    "8b4/pgKHEX2bR9n1Vw5uJRWEtPKVBXeV9dk0D1I/1/iDYJRkc3OwZ1+MYWN31i5r\n" +
    "tg==\n" +
    "-----END PRIVATE KEY-----\n",
  publicKey:
    "-----BEGIN PUBLIC KEY-----\n" +
    "MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQBcsNUnRCg/Us9l6I346gj1aIpauWE\n" +
    "mYywZmbVxtsesnUVFt0PyC+5HxLSsnCeNs/VIT+GnXr8pjQUOCdNv2Mf47MBnCjH\n" +
    "qdvK/NuZFBOuC27FHPG+P6YChxF9m0fZ9VcObiUVhLTylQV3lfXZNA9SP9f4g2CU\n" +
    "ZHNzsGdfjGFjd9Yua7Y=\n" +
    "-----END PUBLIC KEY-----\n",
  type: "EC",
});
