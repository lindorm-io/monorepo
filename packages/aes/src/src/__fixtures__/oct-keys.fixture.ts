import { Kryptos } from "@lindorm/kryptos";

export const OCT_KEY_SET = Kryptos.from("pem", {
  id: "e3ff08aa-af8a-5e3b-9608-6aad72f8fe06",
  privateKey: "aIDH@Vq@FOvVHEngSd@Erjlw@SnsAtMu",
  type: "oct",
});
