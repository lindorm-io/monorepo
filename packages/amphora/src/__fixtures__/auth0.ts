export const OPEN_ID_CONFIGURATION_RESPONSE = {
  issuer: "https://lindorm.eu.auth0.com/",
  authorization_endpoint: "https://lindorm.eu.auth0.com/authorize",
  token_endpoint: "https://lindorm.eu.auth0.com/oauth/token",
  device_authorization_endpoint: "https://lindorm.eu.auth0.com/oauth/device/code",
  userinfo_endpoint: "https://lindorm.eu.auth0.com/userinfo",
  mfa_challenge_endpoint: "https://lindorm.eu.auth0.com/mfa/challenge",
  jwks_uri: "https://lindorm.eu.auth0.com/.well-known/jwks.json",
  registration_endpoint: "https://lindorm.eu.auth0.com/oidc/register",
  revocation_endpoint: "https://lindorm.eu.auth0.com/oauth/revoke",
  scopes_supported: [
    "openid",
    "profile",
    "offline_access",
    "name",
    "given_name",
    "family_name",
    "nickname",
    "email",
    "email_verified",
    "picture",
    "created_at",
    "identities",
    "phone",
    "address",
  ],
  response_types_supported: [
    "code",
    "token",
    "id_token",
    "code token",
    "code id_token",
    "token id_token",
    "code token id_token",
  ],
  code_challenge_methods_supported: ["S256", "plain"],
  response_modes_supported: ["query", "fragment", "form_post"],
  subject_types_supported: ["public"],
  id_token_signing_alg_values_supported: ["HS256", "RS256", "PS256"],
  token_endpoint_auth_methods_supported: [
    "client_secret_basic",
    "client_secret_post",
    "private_key_jwt",
  ],
  claims_supported: [
    "aud",
    "auth_time",
    "created_at",
    "email",
    "email_verified",
    "exp",
    "family_name",
    "given_name",
    "iat",
    "identities",
    "iss",
    "name",
    "nickname",
    "phone_number",
    "picture",
    "sub",
  ],
  request_uri_parameter_supported: false,
  request_parameter_supported: false,
  token_endpoint_auth_signing_alg_values_supported: ["RS256", "RS384", "PS256"],
};

export const OPEN_ID_JWKS_RESPONSE = {
  keys: [
    {
      kty: "RSA",
      use: "sig",
      n: "w8dT6v3E9Lt2vlulAeQJ_AAzYxvMAjKsSsTHBpwjFcojHQzMSneXVAYKIsvZWUvvG5i3NYFFBKROt-ARJmsfk2rkdrdKxqR2jt1Y21aBQ1xwbvhriqkdJ2aDN5M0z_EqEAtT_yU9_OBrPfRA2Q_Frxiitp7Hy94tbAUrdQ07763KoXfQ3-zEoWHvzx0VtIXo9ZJB688V1BfGOVQbOzL0aLfAE1cVKwhEp8jNXNf41XLCzAlqVPF23vvn4kMM4RPf-VrbXAhxnVIrhkeKWPMRYJfqF7BlUJnZtazdO9CRsm5gTY0dLz7TNq6QmjP9qoGdQFU7QReNJgoTUmmcfOIALw",
      e: "AQAB",
      kid: "iPy9pgzr7cFw1kTuiClWE",
      x5t: "Se70fSkOvv-p7UjtI9rNxLR4SD0",
      x5c: [
        "MIIDCTCCAfGgAwIBAgIJOABrHkCZa6X4MA0GCSqGSIb3DQEBCwUAMCIxIDAeBgNVBAMTF29zcHJleS1kZXYuZXUuYXV0aDAuY29tMB4XDTIwMDcwMTA5MTMyMFoXDTM0MDMxMDA5MTMyMFowIjEgMB4GA1UEAxMXb3NwcmV5LWRldi5ldS5hdXRoMC5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDDx1Pq/cT0u3a+W6UB5An8ADNjG8wCMqxKxMcGnCMVyiMdDMxKd5dUBgoiy9lZS+8bmLc1gUUEpE634BEmax+TauR2t0rGpHaO3VjbVoFDXHBu+GuKqR0nZoM3kzTP8SoQC1P/JT384Gs99EDZD8WvGKK2nsfL3i1sBSt1DTvvrcqhd9Df7MShYe/PHRW0hej1kkHrzxXUF8Y5VBs7MvRot8ATVxUrCESnyM1c1/jVcsLMCWpU8Xbe++fiQwzhE9/5WttcCHGdUiuGR4pY8xFgl+oXsGVQmdm1rN070JGybmBNjR0vPtM2rpCaM/2qgZ1AVTtBF40mChNSaZx84gAvAgMBAAGjQjBAMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFASeeUaY9mUPA0MeRpV6sbneZz3YMA4GA1UdDwEB/wQEAwIChDANBgkqhkiG9w0BAQsFAAOCAQEAu3Y6sygzQHsE4CyPFVKRwj5/QF7x83N0nYKxhkS41mrmrenxK83YAz/vpRaTjqbAHpphOXQE5N3pMtgVQJ8JiH9wryHwTKSE8CLA5kC4wraWZMSpwpwXn97NikI6tGzQmbbSxJkjIxhHWm1t4HcMP6tnYlDUG080JY8MOIAdl25VoO8vUlPgccXY19H+c9URTepdnKw/2xUBbGz2ImSSOgvtzeDXgE19S08Qyv9gMskFVp2HXQUASkvv0W4TdBH73gsKC+hx+mJbnjHxyjvBwjewIBLnwLjlimHzshFPNR5hQ19D+6yOo64PNIPfASFqx6E/BtLtxxO+ij7pBmFLrQ==",
      ],
      alg: "RS256",
    },
    {
      kty: "RSA",
      use: "sig",
      n: "3oax-u7-QhCuPpjbQl25oISCrKql2oBb0TEBzT57WiY0yvTVAFInNOiMfwlHvTiDoavIOtJD7YqJSIitYGl1WVU-0yoUCJXLDUw2jhUFfDY-7RGpL-SbTC68AR2L9excS91pO9mdZL94BtsB4mnGUv4JDiCYDQ691wKlhZ3anT8grBTQ1b-jLvH8Cr3peyizTqtHl1xXA5v71xrokXnyO6IBArOy-8Jv0qYIXGkFHMcsfm_wxTg63XgP99ZH0TAf1XGM0N81fZ_q-VTsYK9bn2SfOjrim_dbSTvyAmPddmr_gIFUHOAWl_bDINpVJJJRoekgixuXn6eY3kF1AIuwjw",
      e: "AQAB",
      kid: "IjICkHcf-qq8_stUQ00IN",
      x5t: "NuWYgxSydGi9CoBUNBrHQLc0_so",
      x5c: [
        "MIIDCTCCAfGgAwIBAgIJB/jow8vrbYtSMA0GCSqGSIb3DQEBCwUAMCIxIDAeBgNVBAMTF29zcHJleS1kZXYuZXUuYXV0aDAuY29tMB4XDTIwMDcwMTA5MTMyMFoXDTM0MDMxMDA5MTMyMFowIjEgMB4GA1UEAxMXb3NwcmV5LWRldi5ldS5hdXRoMC5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDehrH67v5CEK4+mNtCXbmghIKsqqXagFvRMQHNPntaJjTK9NUAUic06Ix/CUe9OIOhq8g60kPtiolIiK1gaXVZVT7TKhQIlcsNTDaOFQV8Nj7tEakv5JtMLrwBHYv17FxL3Wk72Z1kv3gG2wHiacZS/gkOIJgNDr3XAqWFndqdPyCsFNDVv6Mu8fwKvel7KLNOq0eXXFcDm/vXGuiRefI7ogECs7L7wm/SpghcaQUcxyx+b/DFODrdeA/31kfRMB/VcYzQ3zV9n+r5VOxgr1ufZJ86OuKb91tJO/ICY912av+AgVQc4BaX9sMg2lUkklGh6SCLG5efp5jeQXUAi7CPAgMBAAGjQjBAMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFJxXf47GEGWjbj/7g+3ZiAc9Vdh+MA4GA1UdDwEB/wQEAwIChDANBgkqhkiG9w0BAQsFAAOCAQEA0/LuunBCvhHF0DczFxBi5vUJ/XTJAtiH8W0MZwftyI19WIrSVHnWyMvoxjsEIlcEMwPXSvcTcH6YMMKmXgceiR8WTx9DEZHdo25pp3GpySXwAwZPDg3AK5DJXv6YUJA4Br7YngDEu+d/ZKOWJHW7Mw8issbijWcuSYDgtSl/kBHBCcTYdiWPN3GHQrOCwwnkQI0yFp8+T2lz9ykiQ+zl8biX2FHRG06qW5NmGYoWyAdPyKbVmjqezmDptKVkCei7cxAH+jAK8SzQ+7CdwHuqCs5jkvhltjgqcex5PRzgEvpVWjPiXrCavq9GZQlWM9nUV2t5B5w4yovdmPhM63dBdA==",
      ],
      alg: "RS256",
    },
  ],
};
