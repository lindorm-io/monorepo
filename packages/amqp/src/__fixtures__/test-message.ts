import { AmqpMessageBase } from "../classes/AmqpMessageBase";

export type TestMessageOptions = {
  name: string;
};

export class TestMessage extends AmqpMessageBase {
  public name: string;

  public constructor(options: TestMessageOptions) {
    super();

    this.name = options.name;
  }
}
