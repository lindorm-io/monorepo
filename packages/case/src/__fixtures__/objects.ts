export const TEST_OBJECT = {
  camelCase: true,
  "Capital Case": true,
  CONSTANT_CASE: true,
  "dot.case": true,
  "kebab-case": true,
  "lower case": true,
  PascalCase: true,
  "path/case": true,
  "Sentence case": true,
  snake_case: true,
};

export const TEST_ARRAY_WITH_OBJECTS = [TEST_OBJECT, TEST_OBJECT];
