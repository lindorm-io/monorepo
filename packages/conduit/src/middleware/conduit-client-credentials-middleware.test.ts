import MockDate from "mockdate";
import nock from "nock";
import { OPEN_ID_CONFIGURATION_RESPONSE } from "../__fixtures__/auth0";
import {
  ConduitClientCredentialsCache,
  ConduitClientCredentialsMiddlewareFactory,
  conduitClientCredentialsMiddlewareFactory,
} from "./conduit-client-credentials-middleware";

const MockedDate = new Date("2024-01-01T00:00:00.000Z");
MockDate.set(MockedDate);

describe("conduit-client-credentials-middleware", () => {
  const cache: ConduitClientCredentialsCache = [];

  let ctx: any;
  let factory: ConduitClientCredentialsMiddlewareFactory;

  const configScope = nock("https://lindorm.eu.auth0.com")
    .get("/.well-known/openid-configuration")
    .times(1)
    .reply(200, OPEN_ID_CONFIGURATION_RESPONSE);

  beforeAll(() => {
    factory = conduitClientCredentialsMiddlewareFactory(
      {
        clientId: "clientId",
        clientSecret: "clientSecret",
        clockTolerance: 0,
        issuer: "https://lindorm.eu.auth0.com/",
      },
      cache,
    );
  });

  beforeEach(() => {
    ctx = {
      req: { headers: {} },
    };
  });

  describe("with audience", () => {
    test("should return a middleware", async () => {
      const tokenScope = nock("https://lindorm.eu.auth0.com")
        .post("/oauth/token")
        .times(1)
        .reply(200, {
          access_token: "access_token",
          expires_in: 10,
          token_type: "Bearer",
        });

      const middleware = await factory({ audience: "https://identity.lindorm-io" });

      expect(configScope.isDone()).toEqual(true);
      expect(tokenScope.isDone()).toEqual(true);

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });

      expect(cache).toMatchSnapshot();
    });

    test("should return middleware with cached data", async () => {
      const middleware = await factory({ audience: "https://identity.lindorm-io" });

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });

      expect(cache).toMatchSnapshot();
    });

    test("should return middleware with timeout data", async () => {
      const scope = nock("https://lindorm.eu.auth0.com")
        .post("/oauth/token")
        .times(1)
        .reply(200, {
          access_token: "access_token",
          expires_in: 10,
          token_type: "Bearer",
        });

      MockDate.set("2024-01-01T00:00:30.000Z");

      const middleware = await factory({ audience: "https://identity.lindorm-io" });

      expect(scope.isDone()).toEqual(true);

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });

      expect(cache).toMatchSnapshot();
    });

    test("should return a middleware with new data", async () => {
      const scope = nock("https://lindorm.eu.auth0.com")
        .post("/oauth/token")
        .times(1)
        .reply(200, {
          access_token: "access_token",
          expires_in: 10,
          token_type: "Bearer",
        });

      const middleware = await factory({ audience: "https://test.lindorm.io" });

      expect(scope.isDone()).toEqual(true);

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });

      expect(cache).toMatchSnapshot();
    });
  });

  describe("without audience", () => {
    test("should return a middleware", async () => {
      const scope = nock("https://lindorm.eu.auth0.com")
        .post("/oauth/token")
        .times(1)
        .reply(200, {
          access_token: "access_token",
          expires_in: 10,
          token_type: "Bearer",
        });

      const middleware = await factory();

      expect(configScope.isDone()).toEqual(true);
      expect(scope.isDone()).toEqual(true);

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });

      expect(cache).toMatchSnapshot();
    });

    test("should return middleware with cached data", async () => {
      const middleware = await factory();

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });

      expect(cache).toMatchSnapshot();
    });
  });

  describe("with internal cache", () => {
    let ctx: any;
    let factory: ConduitClientCredentialsMiddlewareFactory;

    nock("https://lindorm.jp.auth0.com")
      .get("/.well-known/openid-configuration")
      .times(1)
      .reply(200, {
        ...OPEN_ID_CONFIGURATION_RESPONSE,
        token_endpoint: "https://lindorm.jp.auth0.com/oauth/token",
      });

    nock("https://lindorm.jp.auth0.com").post("/oauth/token").times(1).reply(200, {
      access_token: "access_token",
      expires_in: 10,
      token_type: "Bearer",
    });

    beforeAll(() => {
      factory = conduitClientCredentialsMiddlewareFactory({
        clientId: "clientId",
        clientSecret: "clientSecret",
        clockTolerance: 0,
        issuer: "https://lindorm.jp.auth0.com/",
      });
    });

    beforeEach(() => {
      ctx = {
        req: { headers: {} },
      };
    });

    test("should return a middleware", async () => {
      const middleware = await factory();

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });
    });

    test("should return middleware with cached data", async () => {
      const middleware = await factory();

      await expect(middleware(ctx, jest.fn())).resolves.toBeUndefined();

      expect(ctx.req.headers).toEqual({
        Authorization: "Bearer access_token",
      });
    });
  });
});
