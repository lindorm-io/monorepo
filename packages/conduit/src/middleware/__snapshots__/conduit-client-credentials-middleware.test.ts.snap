// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conduit-client-credentials-middleware with audience should return a middleware 1`] = `
[
  {
    "accessToken": "access_token",
    "audience": "https://identity.lindorm-io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067210000,
  },
]
`;

exports[`conduit-client-credentials-middleware with audience should return a middleware with new data 1`] = `
[
  {
    "accessToken": "access_token",
    "audience": "https://identity.lindorm-io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
  {
    "accessToken": "access_token",
    "audience": "https://test.lindorm.io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
]
`;

exports[`conduit-client-credentials-middleware with audience should return middleware with cached data 1`] = `
[
  {
    "accessToken": "access_token",
    "audience": "https://identity.lindorm-io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067210000,
  },
]
`;

exports[`conduit-client-credentials-middleware with audience should return middleware with timeout data 1`] = `
[
  {
    "accessToken": "access_token",
    "audience": "https://identity.lindorm-io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
]
`;

exports[`conduit-client-credentials-middleware without audience should return a middleware 1`] = `
[
  {
    "accessToken": "access_token",
    "audience": "https://identity.lindorm-io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
  {
    "accessToken": "access_token",
    "audience": "https://test.lindorm.io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
  {
    "accessToken": "access_token",
    "audience": "_@DEFAULT",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
]
`;

exports[`conduit-client-credentials-middleware without audience should return middleware with cached data 1`] = `
[
  {
    "accessToken": "access_token",
    "audience": "https://identity.lindorm-io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
  {
    "accessToken": "access_token",
    "audience": "https://test.lindorm.io",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
  {
    "accessToken": "access_token",
    "audience": "_@DEFAULT",
    "issuer": "https://lindorm.eu.auth0.com/",
    "scope": [],
    "tokenType": "Bearer",
    "tokenUri": "https://lindorm.eu.auth0.com/oauth/token",
    "ttl": 1704067240000,
  },
]
`;
